docs/contributor/pocs/gomemlimit-telemetry-memory.md

```diff
@@ -0,0 +1,70 @@
+
+Long-lived memory is something you can estimate upfront or control at runtime. For example, if you have a full-blown cache application, you likely have some sort of limit. Either the cache would stop accepting new values when it's full or start dropping old cache entries. For instance, you could ensure that the cache never exceeds 2GB in size. Then you should be safe on your 4GB machine. The answer is "maybe", but "maybe" is not enough when the risk is running out of memory.
+
+To understand why it is possible to encounter OOM in this situation, we need to look at when the garbage collector runs. We know that we have 2GB of live memory, and simply by using the application, we add a few short-lived heap allocations here and there. We don't expect them to stick around long-term, but since there is no GC cycle running at the moment, they will never be freed. Eventually, we will encounter OOM when intentionally and unintentionally live heap exceeds 4GB.
+
+Now let's look at the other extreme: the Garbage Collector runs extremely frequently. Any time our heap reaches 2.1GB, it runs and removes the 100MB of temporary allocation. An OOM situation is improbable now, but we have far exceeded our cost target; the application might now spend 30-40%, maybe more, on GC. This is no longer efficient.
+
+The optimal solution is the best of two worlds: to get as close to our limit as possible but never beyond it. This way we can delay GC cycles until they are necessary. This will make our application fast, but at the same time, we can be sure that it never crosses the threshold, which makes our application safe from being OOM-killed.
+
+### Understanding Garbage Collection Targets in Go
+We want to make sure we use memory we have without going above it. Before Go 1.19, you had only one knob to turn: the GOGC environment variable. This environment variable accepts a relative target compared to the current live heap size. The default value for GOGC is 100, meaning that the heap should double before GC should run again.
+
+That works well for applications that have small permanent live heaps. For example, if your constant heap is just 50MB and you have 4GB available, you can double your heap targets any time before ever being in danger. If the application load increases and temporary allocation increases, the dynamic targets would be 100MB, 200MB, 400MB, 800MB, 1600MB, and 3200MB. The load must double seven times to cross the 4GB mark, making running out of memory extremely unlikely.
+
+But now think back to our cache application example with a permanent 2GB live heap on a 4GB machine. Even the first doubling of the heap is highly problematic because the new target (4GB) would already reach the limit of the physical memory on the machine.
```

Comment:
Maybe mention that the physical limit could also be a pod or process limit (as in our case)